flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/operators/transform/TableChangeInfo.java

```diff
@@ -139,6 +144,10 @@ public TableChangeInfo deserialize(int version, byte[] serialized) throws IOExce
             try (ByteArrayInputStream bais = new ByteArrayInputStream(serialized);
                     DataInputStream in = new DataInputStream(bais)) {
                 TableId tableId = tableIdSerializer.deserialize(new DataInputViewStreamWrapper(in));
+                if (tableId.equals(magicTableId)) {
+                    version = in.readInt();
+                    tableId = tableIdSerializer.deserialize(new DataInputViewStreamWrapper(in));
+                }
```

Comment:
If no `magicTableId` was present, `version` will not be overwritten and defaults to be latest version (`TableChangeInfo.SERIALIZER.getVersion()`). Is this intended?

Reply:
addressed it.